spec/set-prototype-difference.html

```diff
@@ -4,14 +4,24 @@
     1. Let _set_ be the *this* value.
     1. If Type(_set_) is not Object, throw a *TypeError* exception.
     1. Let _Ctr_ be ? SpeciesConstructor(_set_, %Set%).
-    1. Let _newSet_ be ? Construct(_Ctr_, _set_).
-    1. Let _remover_ be ? Get(_newSet_, "delete").
-    1. If IsCallable(_remover_) is *false*, throw a *TypeError* exception.
-    1. Let _iteratorRecord_ be ? GetIterator(_iterable_).
+    1. Let _iteratorRecord_ be ? GetIterator(_set_).
+    1. If Type(_iterable_) is not Object, throw a *TypeError* exception.
+    1. Let _otherSet_ be _iterable_.
```

Comment:
Should we call `GetIterator` on it here, to effectively assert that it's iterable, and pass that into Construct later?

Reply:
If it has a callable `has`, it should not throw even if it's not iterable. Where do you think it throws?

Reply:
ah i see, true. I guess it seems fine to me to depend on “has” for iterables, but it seems strange to make this method work on a new category, “any object with has” - what’s the motivation to make it this generic?

Reply:
It treats its argument the same way `isSubsetOf` [now does](https://github.com/tc39/proposal-set-methods/pull/41#issuecomment-454205233)., and for the same reason.

(WeakSet is a good example of something which supports `has`, and therefore makes sense as an argument to `isSubsetOf` or `difference`, but which does not support iteration. See that thread, though.)